sycamore/transforms/merge_elements.py

```diff
@@ -0,0 +1,128 @@
+from abc import ABC, abstractmethod
+
+from ray.data import Dataset, ActorPoolStrategy
+
+from sycamore.data import Document, Element, BoundingBox
+from sycamore.plan_nodes import NonCPUUser, NonGPUUser, Transform, Node
+from sycamore.transforms.map import generate_map_class_from_callable
+from sycamore.functions.tokenizer import Tokenizer
+
+from typing import Tuple
+
+
+class ElementMerger(ABC):
+    @abstractmethod
+    def merge_elements(self, document: Document) -> Document:
+        pass
+
+
+class GreedyTextElementMerger(ElementMerger):
+    def __init__(self, tokenizer: Tokenizer, max_tokens: int):
+        self.tokenizer = tokenizer
+        self.max_tokens = max_tokens
+
+    def _should_merge(self, element1: Tuple[Element, int], element2: Tuple[Element, int]) -> bool:
+        if element2[0].type == "Title":
+            return False
+        if element1[1] + 1 + element2[1] > self.max_tokens:
+            return False
+        return True
+
+    def _merge(self, element1: Tuple[Element, int], element2: Tuple[Element, int]) -> Tuple[Element, int]:
+        """
+        Merge two elements; the new element's fields will be set as:
+
+            type: "Section"
+            binary_representation: elt1.binary_representation + elt2.binary_representation
+            text_representation: elt1.text_representation + '\n' + elt2.text_representation
```

Comment:
Why the newline? OpenAI's embedding models actually recommend stripping newlines, though I don't know if it matters for sentence transformers.

Reply:
Just following what Roland did (there should be some kind of separation in case the text_representations are already stripped - we don't want to end up with ".... sword" + "fish ...." = ".... swordfish ...."